diff -burNp transcode-1.1.0/configure.in transcode-1.1.0-dud/configure.in
--- transcode-1.1.0/configure.in	2009-01-17 09:16:36.000000000 +0100
+++ transcode-1.1.0-dud/configure.in	2009-01-29 11:01:30.530815070 +0100
@@ -49,7 +49,6 @@ AC_PATH_PROG(PATH_TO_AWK, $AWK)
 dnl Check the OS.  This has to be done here, after checks for proograms and
 dnl before checks for libraries.
 
-deflib="/lib"
 is_linux=no
 is_bsd=no
 is_osx=no
