--- import/import_mpeg3.c.org	2006-08-20 17:42:22.067601250 +0200
+++ import/import_mpeg3.c	2006-08-20 17:44:33.239799000 +0200
@@ -78,15 +78,16 @@
 
   if (param->flag == TC_VIDEO) {
       if (!file) {
+	  int error_code;
 	  if (!file_a) {
-	      if((file = mpeg3_open(vob->video_in_file))==NULL) {
+	      if((file = mpeg3_open(vob->video_in_file, &error_code))==NULL) {
 		  fprintf(stderr, "open file failed\n");
 		  return(TC_IMPORT_ERROR);
 	      }
 	      if (verbose & TC_DEBUG)
                   printf("[%s] Opened video NO copy\n", MOD_NAME);
 	  } else if (file_a) {
-	      if((file = mpeg3_open_copy(vob->video_in_file, file_a))==NULL) {
+	      if((file = mpeg3_open_copy(vob->video_in_file, file_a, &error_code))==NULL) {
 		  fprintf(stderr, "open file failed\n");
 		  return(TC_IMPORT_ERROR);
 	      }
@@ -97,15 +98,16 @@
   }
   if (param->flag == TC_AUDIO) {
       if (!file_a) {
+	  int error_code;
 	  if (!file) {
-	      if((file_a = mpeg3_open(vob->audio_in_file))==NULL) {
+	      if((file_a = mpeg3_open(vob->audio_in_file, &error_code))==NULL) {
 		  fprintf(stderr, "open audio file failed\n");
 		  return(TC_IMPORT_ERROR);
 	      }
 	      if (verbose & TC_DEBUG)
                   printf("[%s] Opened audio NO copy\n", MOD_NAME);
 	  } else if (file) {
-	      if((file_a = mpeg3_open_copy(vob->audio_in_file, file))==NULL) {
+	      if((file_a = mpeg3_open_copy(vob->audio_in_file, file, &error_code))==NULL) {
 		  fprintf(stderr, "open_copy audio file failed\n");
 		  return(TC_IMPORT_ERROR);
 	      }
