diff -burNp transcode-1.1.0/configure.in transcode-1.1.0-dud/configure.in
--- transcode-1.1.0/configure.in	2009-01-17 09:16:36.000000000 +0100
+++ transcode-1.1.0-dud/configure.in	2009-01-29 11:06:39.880784666 +0100
@@ -1169,7 +1169,7 @@ dnl ImageMagick
 dnl
 IMAGEMAGICK_EXTRA_LIBS="$IMAGEMAGICK_EXTRA_LIBS $($PKG_CONFIG --libs ImageMagick)"
 TC_PKG_CHECK(imagemagick, no, IMAGEMAGICK, no, [magick/api.h],
- none, InitializeMagick, Wand, [http://www.imagemagick.org/])
+ MagickCore, InitializeMagick, MagickWand, [http://www.imagemagick.org/])
 TC_PKG_HAVE(imagemagick, IMAGEMAGICK)
 if test x"$have_imagemagick" = x"yes" ; then
   SAVE_CPPFLAGS="$CPPFLAGS"
