from http://www.freebsd.org/cgi/query-pr.cgi?pr=ports/69406

--- import/decode_dv.c.orig	Thu Jul 22 00:30:32 2004
+++ import/decode_dv.c	Thu Jul 22 00:34:45 2004
@@ -42,6 +42,8 @@
 #define DV_HEADER_SIZE header_size
 
 static int verbose=TC_QUIET;
+static const int frame_size_625_50 = 12 * 150 * 80;
+static const int frame_size_525_60 = 10 * 150 * 80;
 
 static unsigned char *bufalloc(size_t size)
 {
