diff -uNr transcode-0.6.0.orig/libvo/cpu_accel.c transcode-0.6.0/libvo/cpu_accel.c
--- transcode-0.6.0.orig/libvo/cpu_accel.c	Tue Apr  2 12:58:28 2002
+++ transcode-0.6.0/libvo/cpu_accel.c	Tue Aug  6 10:53:58 2002
@@ -115,6 +115,7 @@
 
     canjump = 1;
 
+#if defined(HAVE_PPC_ALTIVEC)
     asm volatile ("mtspr 256, %0\n\t"
 		  "vand %%v0, %%v0, %%v0"
 		  :
@@ -122,6 +123,9 @@
 
     signal (SIGILL, SIG_DFL);
     return MM_ACCEL_PPC_ALTIVEC;
+#else
+    return 0;
+#endif
 }
 #endif /* ARCH_PPC */
 
diff -uNr transcode-0.6.0/libmpeg2/idct_altivec.c transcode-0.6.0.new/libmpeg2/idct_altivec.c
--- transcode-0.6.0/libmpeg2/idct_altivec.c	Tue Apr  2 12:58:28 2002
+++ transcode-0.6.0.new/libmpeg2/idct_altivec.c	Tue Aug  6 11:42:01 2002
@@ -25,7 +25,7 @@
 
 #include "config.h"
 
-#ifdef ARCH_PPC
+#if defined(ARCH_PPC) && defined(HAVE_PPC_ALTIVEC)
 
 #include <inttypes.h>
 
diff -uNr transcode-0.6.0/libmpeg2/motion_comp_altivec.c transcode-0.6.0.new/libmpeg2/motion_comp_altivec.c
--- transcode-0.6.0/libmpeg2/motion_comp_altivec.c	Tue Apr  2 12:58:28 2002
+++ transcode-0.6.0.new/libmpeg2/motion_comp_altivec.c	Tue Aug  6 11:42:33 2002
@@ -25,7 +25,7 @@
 
 #include "config.h"
 
-#ifdef ARCH_PPC
+#if defined(ARCH_PPC) && defined(HAVE_PPC_ALTIVEC)
 
 #include <inttypes.h>
 
diff -uNr transcode-0.6.0.orig/libmpeg2/idct.c transcode-0.6.0/libmpeg2/idct.c
--- transcode-0.6.0.orig/libmpeg2/idct.c	Wed Nov 28 17:50:04 2001
+++ transcode-0.6.0/libmpeg2/idct.c	Tue Aug  6 12:33:14 2002
@@ -74,7 +74,7 @@
 	fprintf (stderr, "Using MMX for IDCT transform\n");
     } else
 #endif
-#ifdef ARCH_PPC
+#if defined(ARCH_PPC) && defined(HAVE_PPC_ALTIVEC)
     if (config.flags & MM_ACCEL_PPC_ALTIVEC) {
 	fprintf (stderr, "Using altivec for IDCT transform\n");
     } else
@@ -103,7 +103,7 @@
 	idct_mmx_init ();
     } else
 #endif
-#ifdef ARCH_PPC
+#if defined(ARCH_PPC) && defined(HAVE_PPC_ALTIVEC)
     if (config.flags & MM_ACCEL_PPC_ALTIVEC) {
 	idct_block_copy = idct_block_copy_altivec;
 	idct_block_add = idct_block_add_altivec;
diff -uNr transcode-0.6.0.orig/libmpeg2/motion_comp.c transcode-0.6.0/libmpeg2/motion_comp.c
--- transcode-0.6.0.orig/libmpeg2/motion_comp.c	Thu Feb 14 20:01:27 2002
+++ transcode-0.6.0/libmpeg2/motion_comp.c	Tue Aug  6 12:34:31 2002
@@ -43,7 +43,7 @@
 	fprintf (stderr, "Using MMX for motion compensation\n");
     } else
 #endif
-#ifdef ARCH_PPC
+#if defined(ARCH_PPC) && defined(HAVE_PPC_ALTIVEC)
     if (config.flags & MM_ACCEL_PPC_ALTIVEC) {
       fprintf (stderr, "Using altivec for motion compensation\n");
     } else
@@ -71,7 +71,7 @@
 	mc_functions = mc_functions_mmx;
     } else
 #endif
-#ifdef ARCH_PPC
+#if defined(ARCH_PPC) && defined(HAVE_PPC_ALTIVEC)
     if (config.flags & MM_ACCEL_PPC_ALTIVEC) {
 	mc_functions = mc_functions_altivec;
     } else
diff -uNr transcode-0.6.0.orig/libdvenc/enc_input.h transcode-0.6.0/libdvenc/enc_input.h
--- transcode-0.6.0.orig/libdvenc/enc_input.h	Thu Dec 27 13:23:31 2001
+++ transcode-0.6.0/libdvenc/enc_input.h	Tue Aug  6 12:45:39 2002
@@ -32,6 +32,7 @@
 extern "C" {
 #endif
 	#define DV_ENC_MAX_INPUT_FILTERS     32
+	#define	DCT_YUV_PRECISION 8
 
 	typedef struct dv_enc_input_filter_s {
 		int (*init)(int wrong_interlace, int force_dct);
